app/services/facet.service.js

```diff
@@ -86,17 +86,20 @@
     var filter = function(values) {
       var self = this; // needs to be defined outside the filter function
       return values.filter(function(element, index, array) {
-        var result;
-        for (var key in element) {
-          var value = element[key];
-          result = filter_value.apply(self, [key, value]);
-          if (result === false) {
-            return false;
-          }
-        }
+        return self.passes_filters(element);
+      });
+    };
 
-        return true;
+    var passes_filters = function(object) {
+      var self = this;
+      var result = true;
+      angular.forEach(object, function(value, key) {
+        if (filter_value.apply(self, [key, value]) === false) {
+          result = false;
```

Comment:
The reason I'm using `angular.forEach` is because it does filtering on the `hasOwnProperty` method, so this doesn't iterate over inherited properties.